### PR TITLE
Include Kotlin DSL samples for Tooling API chapter

### PR DESCRIPTION
**Pull request by eriwen**
This also improves the dependency configuration and plugins{} DSL use for both samples

### Gradle Core Team Checklist
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation
- [ ] Recognize contributor in release notes

### CODE REVIEW

File: subprojects/docs/src/samples/toolingApi/runBuild/kotlin/build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,21 @@
+plugins {
+    application
+}
+
+val toolingApiVersion = gradle.gradleVersion
+
+// tag::use-tooling-api[]
+repositories {
+    maven { url = uri("https://repo.gradle.org/gradle/libs-releases") }
```
Comment (StefMa) [inline subprojects/docs/src/samples/toolingApi/runBuild/kotlin/build.gradle.kts]: I think there is better overload extension for doings so:
`maven(url = "https://repo.gradle.org/gradle/libs-releases")`
See https://github.com/gradle/kotlin-dsl/blob/1d81b63f3a35f4122b6a49b9f57f7381b714b9a5/subprojects/provider/src/main/kotlin/org/gradle/kotlin/dsl/RepositoryHandlerExtensions.kt#L35

File: subprojects/docs/src/samples/toolingApi/runBuild/kotlin/build.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+}
+
+dependencies {
+    implementation("org.gradle:gradle-tooling-api:${toolingApiVersion}")
```
Comment (StefMa) [inline subprojects/docs/src/samples/toolingApi/runBuild/kotlin/build.gradle.kts]: No need for the curly brackets... Just use `$toolingApiVersion`

File: subprojects/docs/src/samples/toolingApi/runBuild/groovy/build.gradle [hunk 3]
```diff
@@ -9,9 +8,9 @@ repositories {
 }
 
 dependencies {
-    compile "org.gradle:gradle-tooling-api:${toolingApiVersion}"
+    implementation "org.gradle:gradle-tooling-api:${toolingApiVersion}"
```
Comment (eskatos) [inline subprojects/docs/src/samples/toolingApi/runBuild/groovy/build.gradle]: :x: no need for curly brackets here

File: subprojects/docs/src/samples/toolingApi/runBuild/kotlin/build.gradle.kts (hunk 1 above)
Comment (eskatos) [reply subprojects/docs/src/samples/toolingApi/runBuild/kotlin/build.gradle.kts]: Yep, the overload can be handy. But let's stick to the more general and more close to Groovy `maven { url = uri("...") }` for user manual snippets. It also has the side effect of demonstrating how to create and assign a URI in the general case.

**Comment by eriwen** [comment]
Thanks all. I cleaned it up and merged.

**Pull request merged**